### PR TITLE
Increase Pillow dependency range to include 10.x

### LINKED ISSUE #10631: Add support for Pillow 10.0.0
### Is your proposal related to a problem?

Pillow 10.0.0 [has just been released.](https://github.com/python-pillow/Pillow/releases/tag/10.0.0) Wagtail 5.0.2 [restricts Pillow support to <10.0.0.](https://github.com/wagtail/wagtail/blob/a68f69f2d7f46943cc23b7f65349448b23044869/setup.py#L30)

Adding support for the new Pillow release is desired.

### Describe the solution you'd like

Add support for Pillow 10.0.0

### Describe alternatives you've considered

Not applicable.

### Additional context

This is a relevant dependency to the project, and to sites running it.

### PR DESCRIPTION
**Pull request by yuekui**
According to the [release notes](https://pillow.readthedocs.io/en/stable/releasenotes/10.0.0.html), the breaking changes in 10.x shouldn't affect the current version of wagtail

Fixes #10631 

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [ ] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by zerolab** [comment]
Willow 1.5 uses "Pillow>=9.1.0,<11.0.0" in testing ([ref](https://github.com/wagtail/Willow/blob/v1.5/setup.py#L53)) and handled the Pillow 10 deprecations in https://github.com/wagtail/Willow/pull/103

File: setup.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "djangorestframework>=3.11.1,<4.0",
     "django-filter>=2.2,<23",
     "draftjs_exporter>=2.1.5,<3.0",
-    "Pillow>=4.0.0,<10.0.0",
+    "Pillow>=4.0.0,<11.0.0",
```
Comment (zerolab) [inline setup.py:30]:
```suggestion
    "Pillow>=9.1.0,<11.0.0",
```

As https://pillow.readthedocs.io/en/stable/releasenotes/index.html mentions

> Please use the latest version of Pillow. Functionality and security fixes should not be expected to be backported to earlier versions.

**Review by zerolab — APPROVED** [review]
Made a small suggestion for more explicit Pillow boundaries.
non-blocking for per se.

**Review by lb- — APPROVED** [review]
Thanks @zerolab  for your feedback before.

Approved - thanks again @yuekui 

A note for future, no need to update the Changelog, it's updated when the core team merge in. We update the release notes and the changelog.

I have included this in the upgrade considerations also.

**Comment by G-Rath** [comment]
Is it possible to get this constraint change backported to v4.1 to allow us to address [PYSEC-2023-175](https://osv.dev/vulnerability/PYSEC-2023-175) while remaining on the current LTS version?

(specifically the widening of the max version, as changing the min is probably a breaking change)

**Comment by zerolab** [comment]
@G-Rath good shout. #10951 should enable that

**Pull request merged**